### PR TITLE
feat: port over /auth

### LINKED ISSUE #445: Port over the functionalities of the /auth endpoint

### PR DESCRIPTION
**Pull request by yunchipang**
## Description

**Issue(s) addressed**

- Resolves #445 

**What kind of change(s) does this PR introduce?**

- [x] Enhancement
- [ ] Bug fix
- [ ] Refactor

**Please check if the PR fulfils these requirements**

- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Issue

**What is the current behavior?**

**What is the new behavior?**

## Breaking change

**Does this PR introduce a breaking change?**

### CODE REVIEW

**Comment by dadiorchen** [comment]
@yunchipang did you finish all test and past them?

**Comment by yunchipang** [comment]
> @yunchipang did you finish all test and past them?

I don't currently see a tests folder on the typescript branch and I'm not sure what the preferred approach is (I was messaging @Kpoke ). Should I create a `__tests__` directory at the root and add a test file for the /auth module there?

If you don't mind, could I slack you for more detailed questions? tysm!

**Comment by yunchipang** [comment]
@dadiorchen @Kpoke tests for root and `auth.service` were finished and passed. i ported over the logic from `__tests__/auth.spec.js` and the test data from `__tests__/seed.js`.
Screenshot 2024-07-18 at 5 52 05 PM
should I write the tests for getting and posting wallets too? or should that be addressed in another issue/PR? tysm🙏🏻

File: src/modules/auth/__tests__/auth.controller.spec.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+  });
+
+  afterAll(async () => {
+    await app.close();
+  });
+
+  it('should login with valid credentials', async () => {
+    const staticApiKey = 'FORTESTFORTESTFORTESTFORTESTFORTEST';
+
+    const response = await request(app.getHttpServer())
```
Comment (Kpoke) [inline src/modules/auth/__tests__/auth.controller.spec.ts:39]: This is more of an integration test than it is a unit test.

Let's have a tests folder in the root folder which will house the integration tests like this https://github.com/Greenstand/treetracker-wallet-api/tree/master/__tests__

and the unit tests can be in the auth folder, like this https://github.com/Greenstand/treetracker-wallet-api/blob/master/server/services/AuthService.spec.js

File: src/modules/utils/event-enum.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export enum AUTH_EVENTS {
```
Comment (Kpoke) [inline src/modules/utils/event-enum.ts:1]: This file can be in the event module

**Pull request merged**